bpf/bpf_overlay.c

```diff
@@ -465,6 +465,8 @@ int tail_handle_ipv4(struct __ctx_buff *ctx)
 	__s8 ext_err = 0;
 	int ret;
 
+	set_identity_mark(ctx, src_sec_identity, MARK_MAGIC_IDENTITY);
+
```

Comment:
Won't this conflict with subsequent code that currently reads from the mark, eg. https://github.com/cilium/cilium/blob/bb947ca474a118fcfdb69f14c10c0f4d34d6cc43/bpf/bpf_overlay.c#L333

I think you might be able to push this further down in the function, to where you actually need it (around the call to `ipv4_host_delivery()`) ?